From f31897f7a3ad9fb1a1e144064bfc8611f598e218 Mon Sep 17 00:00:00 2001
From: Rui Matos <tiagomatos@gmail.com>
Date: Wed, 19 Aug 2015 15:50:29 +0200
Subject: [PATCH] region: If language isn't set in AccountsService, show
 current locale

For newly created user accounts, AccountsService won't have the
language property set. Instead of defaulting to en_US in that case,
let's fallback to the current locale.

https://bugzilla.gnome.org/show_bug.cgi?id=753828
---
 panels/region/cc-region-panel.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/panels/region/cc-region-panel.c b/panels/region/cc-region-panel.c
index e89268b..0955dd7 100644
--- a/panels/region/cc-region-panel.c
+++ b/panels/region/cc-region-panel.c
@@ -614,11 +614,12 @@ static void
 update_language_from_user (CcRegionPanel *self)
 {
 	CcRegionPanelPrivate *priv = self->priv;
-        const gchar *language;
+        const gchar *language = NULL;
 
         if (act_user_is_loaded (priv->user))
                 language = act_user_get_language (priv->user);
-        else
+
+        if (language == NULL || *language == '\0')
                 language = setlocale (LC_MESSAGES, NULL);
 
         g_free (priv->language);
-- 
2.4.3

